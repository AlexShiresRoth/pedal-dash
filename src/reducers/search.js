import { SEARCH_PEDALS } from '../actions/types';
const initialState = {
	pedalRes: [],
	loading: true,
};

export default (state = initialState, action) => {
	const { type, payload } = action;
	switch (type) {
		case SEARCH_PEDALS:
			return {
				...state,
				pedalRes: payload,
				loading: false,
			};
		default:
			return state;
	}
};
