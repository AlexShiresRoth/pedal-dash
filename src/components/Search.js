import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { MdSearch, MdClose } from 'react-icons/md';
import { connect } from 'react-redux';
import { getPedals, clearData } from '../actions/search';
import searchStyles from './searchstyles/Search.module.scss';

const Search = ({ getPedals, clearData }) => {
	const [formData, setFormData] = useState({
		query: '',
	});

	const { query } = formData;

	const onChange = e => {
		setFormData({
			...formData,
			[e.target.name]: e.target.value,
		});
	};

	const onSubmit = e => {
		e.preventDefault();
		getPedals(formData);
	};

	const clearSearch = e => {
		e.preventDefault();

		setFormData({
			query: '',
		});
		clearData();
	};

	return (
		<form onSubmit={e => onSubmit(e)} className={searchStyles.form}>
			<div className={searchStyles.input__row}>
				<button onSubmit={e => onSubmit(e)}>
					<MdSearch />
				</button>
				<input
					data-testid="search-query"
					type="text"
					value={query}
					name="query"
					onChange={e => onChange(e)}
					placeholder="Earthquaker Devices"
					required
				></input>
				<button onClick={e => clearSearch(e)}>
					<MdClose />
				</button>
			</div>
		</form>
	);
};

Search.propTypes = {
	getPedals: PropTypes.func.isRequired,
};

export default connect(null, { getPedals, clearData })(Search);
