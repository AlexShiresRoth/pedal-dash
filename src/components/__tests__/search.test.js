import React from 'react';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import { initialState, reducers } from '../../reducers';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Search from '../Search';

afterEach(cleanup);

function renderWithRedux(ui, { initialState, store = createStore(combineReducers({ reducers, initialState })) } = {}) {
	return {
		...render(<Provider store={store}>{ui}</Provider>),
		store,
	};
}

test('can render component with redux', () => {
	const searchCont = renderWithRedux(<Search />);
	expect(searchCont.firstChild).toMatchSnapshot();
});

test('can render with redux with custom initial state', () => {
	const { getByTestId } = renderWithRedux(<Search />, {
		initialState: { query: '' },
	});
	expect(getByTestId('search-query')).toHaveTextContent('');
});
