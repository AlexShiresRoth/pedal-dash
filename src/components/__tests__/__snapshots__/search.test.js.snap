// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render component with redux 1`] = `undefined`;

exports[`can render with redux 1`] = `undefined`;
